### PR TITLE
573 Story to Create Quiz Service

### LINKED ISSUE #573: Story: Create Quiz Service
# Create Quiz Service

As a **developer**, I want **quizes to be database driven**, so that **We can removed them from the code base and keep them in persistent storage**.

Associated Epic: #566

## Definition of Done

- [x] Acceptance criteria defined
- [x] Solution tasks specified
- [x] Feature branch created
- [x] ~~Unit tests written (on API code only for now)~~
- [ ] ~~Solution passes full suite of unit tests~~
- [ ] Design documents updated if needed
- [ ] Pull request created
- [ ] Code is reviewed
- [ ] Code is merged to `develop` branch
- [ ] ~~Tests pass on CI~~
- [ ] ~~Deployed to staging environment~~
- [ ] Acceptance criteria is met

## Acceptance Criteria

- [x] Given **a user**, when **they start the quiz**, then **the quiz properly populates**
- [x] Given **a developer**, when **sends a request to the /quiz/id route**, then **they receive the quiz body**
- [x] Given **a user**, when **they enter a multipart quiz**, then **quiz handler should properly set the states of the quiz**

## Solution Tasks

- [x] Create Quiz Service
- [x] Create Quiz Controller
- [ ] have the quiz handler make the request and state properly

### PR DESCRIPTION
**Pull request by AndreasLc1103**
# 573 Story to Create Quiz Service
closes #573 

## Description 
This story is intended on finalizing the ability to making the quiz database driven and not dependent on the API directory., eliminating the need for extra files storing json within the codebase

### User Story
- As a **Developer** I would like to **have the quiz database driven** so **we can update a single source of truth**

- [x] No discrepancies across browsers. Eg. chrome vs safari
- [x] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [x] CSS (https://jigsaw.w3.org/css-validator/)
- [x] Google sign-in is operational
- [x] Accessibility functions
- [x] Pages can scale without distorting page
- [x] No dead links
- [x] Navbar is consistent across the site
- [x] Pages are screen-reader accessible
- [x] Contrast meets standards for accessibility
- [x] Pages are keyboard accessible
- [x] Passes WAVE Evaluation Tool chrome extension

### Testing instructions 

1. cd into database directory and drop the db container using `` docker compose down -v``
2. bring back up with ``docker compose up``
3. cd into the server directory and run ``npm test`` and ensure tests pass 
4. then use pm2 or manually run the setup 
5. go through the quizzes on the site

### CODE REVIEW

**Review by dm9965 — APPROVED** [review]
Great job cleaning up a ton of static code bloat that should've been in the db to begin with. Also updating the Lab Service to include updated, more secure methods is a good touch. Good job on the jest tests as well.

File: client/src/components/exercise/lab7/pages/AlterationActivity/AlterationQuiz.js [hunk 1]
```diff
@@ -13,6 +15,7 @@ import useMainStateContext from "src/reducers/MainContext";
  */
 const AlterationQuiz = () => {
   const [showContinue, setShowContinue] = useState(false);
+  // eslint-disable-next-line no-unused-vars
```
Comment (heathermoses) [inline client/src/components/exercise/lab7/pages/AlterationActivity/AlterationQuiz.js]: nit: remove eslint

**Review by heathermoses — APPROVED** [review]
This looks good overall. I requested one small change, I'm approving in the meantime

When testing some of the quizzes, I got an undefined type error. Refreshing the page made the error go away, but I'm not sure why that would have happened
Screenshot 2024-04-22 at 10 52 53 AM

**Comment by AndreasLc1103** [comment]
> This looks good overall. I requested one small change, I'm approving in the meantime
> 
> When testing some of the quizzes, I got an undefined type error. Refreshing the page made the error go away, but I'm not sure why that would have happened Screenshot 2024-04-22 at 10 52 53 AM

resolved

**Review by heathermoses — APPROVED** [review]

**Pull request merged**